### PR TITLE
RD-6920 Terraform modal refactoring

### PR DESCRIPTION
**Pull request by kubama**
## Description
Terraform modal refactoring

## Screenshots / Videos
N/A

## Checklist

- [X] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [X] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/4677/
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/4686/

 

## Documentation
N/A

### CODE REVIEW

**Review by PatrycjaKubiczek — APPROVED** [review]
LGTM 👍

File: app/widgets/common/terraformModal/TerraformModal.tsx [hunk 1]
```diff
@@ -203,25 +159,27 @@ function markDuplicates(
     setVariables(variables);
 }
 
+type ErrorSetter = (path: string, error: any) => void;
```
Comment (qooban) [inline app/widgets/common/terraformModal/TerraformModal.tsx:162]: Do you think it's achievable to replace `any` with proper type here?

**Review by qooban — COMMENTED** [review]
Cool refactoring 👍🏻

File: app/widgets/common/terraformModal/TerraformModal.tsx (hunk 1 above)
Comment (kubama) [reply app/widgets/common/terraformModal/TerraformModal.tsx]: Done

File: app/utils/hooks/useFormErrors.ts [hunk 2]
```diff
@@ -1,26 +0,0 @@
-import { useDispatch, useSelector } from 'react-redux';
-import { useCallback } from 'react';
-import type { ReduxState } from '../../reducers';
-import { setFieldError, cleanFormErrors } from '../../actions/formErrors';
-import type { FieldErrors } from '../../reducers/formErrorsReducer';
-
-export default function useFormErrors(formName: string) {
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/utils/hooks/useFormErrors.ts:7]: Looks like no longer used, so removed

**Review by qooban — APPROVED** [review]
I see you removed new test. Intentionally? 🤔

**Comment by kubama** [comment]
> I see you removed new test. Intentionally? thinking

It wasn't new, somehow it got duplicated, so I removed just the copy

**Pull request merged**